extension/amqp/amqp.go

```diff
@@ -0,0 +1,54 @@
+package amqp
+
+import (
+	"io"
+
+	"github.com/hellofresh/goengine"
+	"github.com/streadway/amqp"
+)
+
+// setup returns a connection and channel to be used for consumption with the Queue setup
+func setup(url, queue string) (*amqp.Connection, *amqp.Channel, error) {
+	conn, err := amqp.Dial(url)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ch, err := conn.Channel()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if _, err := ch.QueueDeclare(queue, true, false, false, false, nil); err != nil {
+		return nil, nil, err
+	}
+
+	return conn, ch, nil
+}
+
+// DirectQueueConsume returns a Consume func that will connect to the provided AMQP server and create a queue for direct message delivery
+func DirectQueueConsume(amqpDSN, queue string) (Consume, error) {
+	switch {
+	case len(amqpDSN) == 0:
```

Comment:
Shouldn't we do `amqp.ParseURI()` instead?

Reply:
I wanted to disagree but it makes way to much sense to fail early